### PR TITLE
refactor(core): Reactify ExecutionWindowDayPicker component

### PR DESCRIPTION
**Pull request by Jammy-Louie**
Currently refactoring ExecutionWindow to React. First PR is to refactor ExecutionWindowDayPicker to React before I complete the rest

### CODE REVIEW

**Comment by Jammy-Louie** [comment]
Refactored executionWindows
Screen Shot 2019-06-14 at 4 37 03 PM

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindowDayPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  const none = (): void => {
+    props.onChange([]);
+  };
+
+  const weekdays = (): void => {
+    props.onChange([2, 3, 4, 5, 6]);
+  };
+
+  const weekend = (): void => {
+    props.onChange([2, 3, 4, 5, 6]);
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindowDayPicker.tsx]:
```suggestion
    props.onChange([1, 7]);
```

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+  const windowDaysUpdated = (newDays: number[]): void => {
+    const restrictedExecutionWindow = {
+      ...props.restrictedExecutionWindow,
+      days: newDays,
+    };
+    props.updateStageField({ restrictedExecutionWindow });
+  };
+
+  const updateWhiteList = (changes: Partial<IWindow>, index: number) => {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: super minor: we are not capitalizing `List` in `whitelist` elsewhere.

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+        windows.push(buildTimelineWindow(firstWindow, window));
+        windows.push(buildTimelineWindow(secondWindow, window));
+      } else {
+        windows.push(buildTimelineWindow(window));
+      }
+    });
+    return windows;
+  }
+
+  function buildTimelineWindow(window: IWindow, originalWindow?: IWindow): ITimeLineWindow {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: same super minor capitalization comment: favor `ITimelineWindow` over `ITimeLineWindow`

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+              </p>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <div className="execution-window-graph">
+                <div className="execution-day">
+                  {timelineWindows.map((w, i) => {
+                    return (
+                      <Tooltip key={i} value={moment(w.start).format('HH:mm') + ' - ' + moment(w.end).format('HH:mm')}>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: We moved off momentjs a few months ago (I think it's still being brought in by some transient dependencies, which is why it's available here). Can you switch this to either `date-fns` or `luxon`? Both have a decent `format` method that should be basically compatible.

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+          {whitelist.map((w: IWindow, i: number) => {
+            return (
+              <div className="window-entry" key={i}>
+                <div className="form-group">
+                  <div className="col-md-6 col-md-offset-1">
+                    <span className="start-end-divider">From</span>
+                    <ReactSelectInput
+                      clearable={false}
+                      inputClassName={
+                        'visible-xs-inline-block visible-sm-inline-block visible-md-inline-block visible-lg-inline-block vertical-align-select'
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: This class string gets repeated a few times below, and it's pretty long - it's probably worth moving it up to a variable and referencing that.

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+                value={enableCustomSkipWindowText}
+                onChange={() => setEnableCustomSkipWindowText(!enableCustomSkipWindowText)}
+              />
+            </div>
+            {enableCustomSkipWindowText && (
+              <div className="col-md-10 col-md-offset-1 checkbox-padding">
+                <TextAreaInput
+                  rows={4}
+                  onChange={(e: React.ChangeEvent<any>) => props.updateStageField({ skipWindowText: e.target.value })}
+                  placeholder={"Default text: '" + defaultSkipWindowText + "' (HTML is okay)"}
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]:
```suggestion
                  placeholder={`Default text: '${defaultSkipWindowText}' (HTML is okay)`}
```

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindowDayPicker.tsx (hunk 1 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindowDayPicker.tsx]: whoops, nice catch. Thanks

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 3 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: :+

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 4 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: 👍

**Comment by Jammy-Louie** [comment]
Changes made, Thanks!

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 2 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: 👍

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 3 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: 👍

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 5 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: 👍

File: app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx (hunk 6 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx]: 👍

**Pull request merged**